src/Microsoft.Restier.Core/Conventions/ConventionBasedEntitySetFilter.cs

```diff
@@ -22,23 +22,33 @@ private ConventionBasedEntitySetFilter(Type targetType)
             this.targetType = targetType;
         }
 
+        public IQueryExpressionFilter Inner { get; set; }
+
         /// <inheritdoc/>
         public static void ApplyTo(
             IServiceCollection services,
             Type targetType)
         {
             Ensure.NotNull(services, "services");
             Ensure.NotNull(targetType, "targetType");
-            services.CutoffPrevious<IQueryExpressionFilter>(new ConventionBasedEntitySetFilter(targetType));
+            services.ChainPrevious<IQueryExpressionFilter>(next => new ConventionBasedEntitySetFilter(targetType)
```

Comment:
Just a proposal: we can access the target type inside `ConventionBaseEntitySetFilter` thus no need to pass in `targetType` and use this overload of `ChainPrevious<TService>`.